java-manta-it/src/test/java/com/joyent/manta/client/MantaClientDirectoriesIT.java

```diff
@@ -242,4 +242,93 @@ private void directorySkipping(final boolean settingCorrectDepth) throws IOExcep
         }
     }
 
+    
+    
+    /**
+     * This test will create a set of directories, then it will 
+     * set the pruneEmptyParentDepth = -1, meaning it will 
+     * delete all empty directories in the hierarchy.
+     * @throws IOException
+     */
+    @Test
+    public void pruneParentDirectoriesFull() throws IOException {
+        final String parentDir = createRandomDirectory(testPathPrefix, 1);
+        // We are going to create a sibling to the parent directory, so 
+        // the test does not delete the root.
+        createRandomDirectory(testPathPrefix, 1);
+        final String childDir = createRandomDirectory(parentDir, 5);
+        LOG.info("CHILD DIR  : " + childDir);
+        LOG.info("Parent DIR : " + parentDir);
+        mantaClient.delete(childDir, null, PruneEmpytParentDirectoryStrategy.PRUNE_ALL_PARENTS);
+        MantaAssert.assertResponseFailureStatusCode(404, RESOURCE_NOT_FOUND_ERROR,
+                (MantaFunction<Object>) () -> mantaClient.get(childDir));
+        MantaAssert.assertResponseFailureStatusCode(404, RESOURCE_NOT_FOUND_ERROR,
+                (MantaFunction<Object>) () -> mantaClient.get(parentDir));
+        // Getting the path of the parent's parent.
+        String ancestor = parentDir.substring(0,parentDir.lastIndexOf(SEPARATOR));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(ancestor));
+    }
+
+    @Test
+    public void pruneParentDirectoryZero() throws IOException {
+        final String parentDir = createRandomDirectory(testPathPrefix, 1);
+        final String childDir = createRandomDirectory(parentDir, 2);
+        LOG.debug("CHILD DIR  : " + childDir);
+        LOG.debug("Parent DIR : " + parentDir);
+        mantaClient.delete(childDir, null, 0);
+        MantaAssert.assertResponseFailureStatusCode(404, RESOURCE_NOT_FOUND_ERROR,
+                (MantaFunction<Object>) () -> mantaClient.get(childDir));
+        Assert.assertTrue(mantaClient.existsAndIsAccessible(parentDir));
+    }
+
+    @Test
+    public void pruneParentDirectoryOne() throws IOException {
+        // This should stop at 1 parent being deleted
+        final String parentDir = createRandomDirectory(testPathPrefix, 1);
+        final String childDir = createRandomDirectory(parentDir, 3);
+        LOG.info("CHILD DIR  : " + childDir);
+        LOG.info("Parent DIR : " + parentDir);
+        mantaClient.delete(childDir, null, 1);
+        MantaAssert.assertResponseFailureStatusCode(404, RESOURCE_NOT_FOUND_ERROR,
```

Comment:
Not a necessary change but these can probably be simplified to `Assert.assertFalse(mantaClient.existsAndIsAccessible(childDir)`.

Up to you if you want to make this change.